pkg/util/logger_windows.go

```diff
@@ -0,0 +1,117 @@
+package util
+
+import (
+	"runtime"
+	"strings"
+
+	"github.com/weaveworks/common/logging"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/go-kit/kit/log"
+	"github.com/weaveworks/common/server"
+	el "golang.org/x/sys/windows/svc/eventlog"
+)
+
+// Default name for the Grafana Agent under Windows
+const ServiceName = "Grafana Agent"
+
+//NewWindowsEventLogger creates a new logger that writes to the event log
+func NewWindowsEventLogger(cfg *server.Config) *Logger {
+	return newLogger(cfg, makeWindowsEventLogger)
+}
+
+func makeWindowsEventLogger(cfg *server.Config) (log.Logger, error) {
+	// Setup the log in windows events
+	err := el.InstallAsEventCreate(ServiceName, el.Error|el.Info|el.Warning)
+
+	// We expect an error of 'already exists' for subsequent runs,
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return nil, err
+	}
+	il, err := el.Open(ServiceName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Cleanup the handle when exits scope, this could be handled via explicit close but would need to change
+	// more upstream, and its not a huge issue if it hangs around slightly longer than it should
```

Comment:
nit: "not a huge issue" makes it sound like it might be a tiny issue :) Can you re-write this to clarify that it's valid to have more than one event logger open concurrently?

```suggestion
	// Ensure the logger gets closed when the GC runs. It's valid to have more than one win logger open concurrently. 
```